### PR TITLE
[brew] give option to run commands in iTerm instead of Terminal

### PR DESCRIPTION
**Pull request by urwrstkn8mare**
## Description

Basically what it says in the title. 

I also made it so that no matter if it is iTerm or Terminal it will focus on the window and exit (must change a setting in Terminal for it to close automatically) after the command is finished and the user presses a key.

~~The actual commit for this PR is https://github.com/raycast/extensions/commit/5a4eae238e7089f38e60254cdbd7d98ab6c8308d but I'm guessing they will be squashed so it should be fine. The file changes should be accurate as I synced with upstream. Sorry I accidentally forgot that I did my previous PR on the main branch leaving the raw commits behind.~~

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/brew/src/components/runInTerminal.ts [hunk 1]
```diff
@@ -1,10 +1,17 @@
...
   runAppleScript(`
-    tell application "Terminal"
-      do script "${command}"
+    tell application "${preferences.useIterm2 ? "iTerm" : "Terminal"}"
+      ${
+        preferences.useIterm2
+          ? "set newWindow to create window with default profile command"
+          : "do shell script \"open -a 'Terminal'\"\ndo script"
+      } "bash -c '${command}; read -n 1 -s -r -p \\"Press any key to exit - will not quit\\" ; echo' ; exit" ${
+    preferences.useIterm2 ? "" : "in selected tab of the front window"
```
Comment (nhojb) [inline extensions/brew/src/components/runInTerminal.ts:13]: The apple script is getting a bit messy and hard to follow here. I'd suggest breaking out into two separate invocations, one for Terminal and one for iTerm.

**Review by nhojb — CHANGES_REQUESTED** [review]
Hey @urwrstkn8mare, thanks for the contribution! This looks really useful.

I think an improvement in terms of UX would be to make the "Terminal" preference a `dropdown` with options for "Terminal" and "iTerm". This would make the preference a little clearer and also will allow adding additional terminal emulators in the future.

File: extensions/brew/package.json [hunk 2]
```diff
@@ -56,6 +57,23 @@
       "required": false,
       "title": "Custom Brew Executable Path",
       "description": "Set this if your brew executable is in a non-standard location"
+    },
+    {
+      "name": "terminalApp",
+      "type": "dropdown",
+      "required": false,
+      "title": "Choose Terminal Application",
```
Comment (nhojb) [inline extensions/brew/package.json:65]: Maybe just "Terminal Application" for the title?

File: extensions/brew/package.json [hunk 3]
```diff
@@ -56,6 +57,23 @@
...
+        {
+          "title": "Terminal",
+          "value": "terminal"
+        },
+        {
+          "title": "iTerm",
+          "value": "iterm"
+        }
+      ],
+      "description": "Choose the terminal application/emulator to use when using any Run in Terminal action."
```
Comment (nhojb) [inline extensions/brew/package.json:76]: Also here: "The terminal application..."

File: extensions/brew/package.json (hunk 2 above)
Comment (urwrstkn8mare) [reply extensions/brew/package.json:65]:
```suggestion
      "title": "Terminal Application",
```
👍

File: extensions/brew/package.json (hunk 3 above)
Comment (urwrstkn8mare) [reply extensions/brew/package.json:76]:
```suggestion
      "description": "The terminal application/emulator to use when using any Run in Terminal action."
```
👍

**Review by nhojb — APPROVED** [review]
